File: java/matrixRain.java
import javax.swing.*;
import java.awt.*;
import java.util.Random;

public class matrixRain extends JPanel {
    private final int fontSize = 16;
    private final String chars = "0123456789/*-+/<>?;:[]~!@#$%^&*()+=abcdefghijklmnopqrstuvwxyz";
    private final int columns;
    private final int[] drops;
    private final int[] nameDrops;
    private final Random random;
    private final Timer timer;
    private boolean nameFallen = false;
    private final String authorName = "Nerrison";

    public matrixRain() {
        Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
        int width = screenSize.width;
        int height = screenSize.height;

        columns = width / fontSize;
        drops = new int[columns];
        nameDrops = new int[columns]; // Array for author's name drops
        random = new Random();

        setPreferredSize(new Dimension(width, height));
        setBackground(Color.BLACK);
        setDoubleBuffered(true); // Enable double buffering

        // Timer to control the animation frame rate
        timer = new Timer(50, e -> repaint());
        timer.start();
    }

    @Override
    protected void paintComponent(Graphics g) {
        super.paintComponent(g);
        Graphics2D g2d = (Graphics2D) g;

        // Draw a semi-transparent background to create a trailing effect
        g2d.setComposite(AlphaComposite.getInstance(AlphaComposite.SRC_OVER, 0.05f));
        g2d.setColor(Color.BLACK);
        g2d.fillRect(0, 0, getWidth(), getHeight());

        if (!nameFallen) {
            // Draw the author's name falling from the top
            g2d.setComposite(AlphaComposite.getInstance(AlphaComposite.SRC_OVER, 1f));
            g2d.setColor(Color.GREEN);
            g2d.setFont(new Font("Monospaced", Font.PLAIN, fontSize));

            for (int x = 0; x < columns; x++) {
                int y = nameDrops[x] * fontSize;
                char text = authorName.charAt(x % authorName.length()); // Use modulus to cycle through the name
                g2d.drawString(String.valueOf(text), x * fontSize, y);

                if (y >= getHeight() || random.nextFloat() > 0.975) {
                    nameDrops[x] = 0;
                } else {
                    nameDrops[x]++;
                }
            }

            // Check if all parts of the name have fallen
            if (nameDrops[0] * fontSize >= getHeight()) {
                nameFallen = true;
            }
        } else {
            // Draw the characters for Matrix Rain
            g2d.setComposite(AlphaComposite.getInstance(AlphaComposite.SRC_OVER, 1f));
            g2d.setColor(Color.GREEN);
            g2d.setFont(new Font("Monospaced", Font.PLAIN, fontSize));

            for (int x = 0; x < columns; x++) {
                int y = drops[x] * fontSize;
                char text = chars.charAt(random.nextInt(chars.length()));
                g2d.drawString(String.valueOf(text), x * fontSize, y);

                if (y >= getHeight() || random.nextFloat() > 0.975) {
                    drops[x] = 0;
                } else {
                    drops[x]++;
                }
            }
        }
    }

    public static void main(String[] args) {
        JFrame frame = new JFrame("Matrix Rain Effect");
        matrixRain matrixRain = new matrixRain();

        // Set a medium window size
        frame.setSize(800, 600);

        frame.add(matrixRain);
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.setLocationRelativeTo(null); // Center the window on the screen
        frame.setVisible(true);
    }
}

File: python/matrix_rain.py
import pygame
import random
import sys

# Initialize pygame
pygame.init()

# Screen dimensions (set to medium size)
WIDTH, HEIGHT = 800, 600
FONT_SIZE = 16
columns = WIDTH // FONT_SIZE

# Colors
BLACK = (0, 0, 0)
GREEN = (0, 255, 0)

# Initialize screen
screen = pygame.display.set_mode((WIDTH, HEIGHT))
pygame.display.set_caption('Matrix Rain Effect')
screen.fill(BLACK)

# Set up font
font = pygame.font.Font(None, FONT_SIZE)

# Create a list to hold the positions of the drops
drops = [0] * columns

def draw_matrix():
    screen.fill(BLACK)  # Clear the screen with black background

    for x in range(columns):
        char = random.choice("0123456789/*-+/<>?;:[]~!@#$%^&*()+=abcdefghijklmnopqrstuvwxyz")
        y = drops[x] * FONT_SIZE
        text = font.render(char, True, GREEN)
        screen.blit(text, (x * FONT_SIZE, y))

        if y > HEIGHT or random.random() > 0.975:
            drops[x] = 0
        else:
            drops[x] += 1

    pygame.display.flip()  # Update the screen

def main():
    clock = pygame.time.Clock()

    while True:
        for event in pygame.event.get():
            if event.type == pygame.QUIT:
                pygame.quit()
                sys.exit()

        draw_matrix()
        clock.tick(30)  # Frame rate

if __name__ == '__main__':
    main()
